lib/redis_memo/memoize_method.rb

```diff
@@ -49,15 +67,19 @@ def self.method_id(ref, method_name)
     "#{class_name}#{is_class_method ? '::' : '#'}#{method_name}"
   end
 
+  def self.extract_dependencies(ref, *method_args, &depends_on)
```

Comment:
Similar to how rails caches each query, we can also apply some request level caching for this:
```
cached_dependencies[ref][method_args] ||= ...
```

Reply:
Were you thinking we should have a separate request level cache as a thread local variable, similar to the `RedisMemo::Cache.local_cache` for caching redis calls?

E.g. if we have 
```
memoize_method :method_1 do ...
  depends_on memo
  depends_on memo
end

memoize_method :method_2 do ...
  depends_on dependency_of(:method_1, *args)
end
```
and within one request we call `method_2` and `method_1`, then the computation to extract method_1's dependencies only happen once?

Reply:
> and within one request we call `method_2` and `method_1`, then the computation to extract method_1's dependencies only happen once?

Indeed!


> Were you thinking we should have a separate request level cache as a thread local variable, similar to the `RedisMemo::Cache.local_cache` for caching redis calls?

Hmm I think we should just use the local_cache so it can flushed automatically every request. The number of things we're storing in memory might be pretty big so we might not be able to persistent them to serve different requests.